tests/models/edx/test_navigational.py

```diff
@@ -0,0 +1,190 @@
+"""Tests for the navigational event models"""
+
+import json
+import re
+
+import pytest
+from hypothesis import given, provisional, settings
+from hypothesis import strategies as st
+from pydantic.error_wrappers import ValidationError
+
+from ralph.models.edx.base import BaseContextField
+from ralph.models.edx.navigational import (
+    NavigationalEventField,
+    UIPageClose,
+    UISeqGoto,
+    UISeqNext,
+    UISeqPrev,
+)
+from ralph.models.selector import ModelSelector
+
+
+@settings(max_examples=1)
+@given(st.builds(UIPageClose, referer=provisional.urls(), page=provisional.urls()))
+def test_models_edx_ui_page_close_selector_with_valid_event(event):
+    """Tests given a page_close event the get_model method should return UIPageClose model."""
+
+    event = json.loads(event.json())
+    assert ModelSelector(module="ralph.models.edx").get_model(event) is UIPageClose
+
+
+@settings(max_examples=1)
+@given(st.builds(NavigationalEventField))
+def test_fields_edx_navigational_events_event_field_with_valid_content(field):
+    """Tests that a valid NavigationalEventField field does not raise a ValidationError."""
+
+    assert re.match(
+        r"^block-v1:[^\/+]+(\/|\+)[^\/+]+(\/|\+)[^\/?]+type@sequential\+block@[a-f0-9]{32}$",
+        field.id,
+    )
+
+
+@pytest.mark.parametrize(
+    "id",
+    [
+        (
+            "block-v2:universityX=CS111+2020_T1+"
+            "type@sequential+block@d0d4a647742943e3951b45d9db8a0ea1",
+        ),
+        (
+            "block-v1:universityX=CS1112020_T1+"
+            "type@sequential+block@d0d4a647742943e3951b45d9db8a0ea1",
+        ),
+        (
+            "block-v1:universityX=CS111=2020_T1+"
+            "tipe@sequential+block@d0d4a647742943e3951b45d9db8a0ea1",
+        ),
+        (
+            "block-v1:universityX=CS111=2020_T1+"
+            "type@sequentialblock@d0d4a647742943e3951b45d9db8a0ea1",
+        ),
+        (
+            "block-v1:universityX=CS111=2020_T1+"
+            "type@sequential+block@d0d4a647742943z3951b45d9db8a0ea1",
+        ),
+        (
+            "block-v1:universityX=CS111=2020_T1+"
+            "type@sequential+block@d0d4a647742943e3951b45d9db8a0ea13",
+        ),
+    ],  # pylint: disable=invalid-name
+)
```

Comment:
Optionally, by shortening the names it's possible to fit each test case in one line:
Exemple:
```suggestion
@pytest.mark.parametrize(
    "id",
    [
        "block-v2:orgX=CS111+20_T1+type@sequential+block@d0d4a647742943e3951b45d9db8a0ea1",
        "block-v1:orgX=CS11120_T1+type@sequential+block@d0d4a647742943e3951b45d9db8a0ea1",
        "block-v1:orgX=CS111=20_T1+tipe@sequential+block@d0d4a647742943e3951b45d9db8a0ea1",
        "block-v1:orgX=CS111=20_T1+",
        "type@sequentialblock@d0d4a647742943e3951b45d9db8a0ea1",
        "block-v1:orgX=CS111=20_T1+type@sequential+block@d0d4a647742943z3951b45d9db8a0ea1",
        "block-v1:orgX=CS111=20_T1+type@sequential+block@d0d4a647742943e3951b45d9db8a0ea13",
    ],  # pylint: disable=invalid-name
)
```